compiler/rustc_ast/src/ast.rs

```diff
@@ -801,8 +801,8 @@ pub enum PatKind {
     /// A reference pattern (e.g., `&mut (a, b)`).
     Ref(P<Pat>, Mutability),
 
-    /// A literal.
-    Lit(P<Expr>),
+    /// A literal, const block or path.
+    Expr(P<Expr>),
```

Comment:
@ytmimi this is the extent of the current PRs changes to the ast

Reply:
Right, but presumably the `PatKind::Lit(P<Expr>)` always held an `ExprKind::Lit(_)`. The new name makes that ambiguous, and I'd like to update the match in rustfmt to be more explicit. Is that possible? If not in this PR can we do that in a follow up soon?